src/Components/Feed/Posts.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react'
+import { Post } from './Post.jsx'
+import axios from 'axios';
+import { WithPost } from '../../HOCs/WithPost.jsx';
+import { useDoLike } from '../apis/useDoLike.jsx';
+import { useGetPost } from '../apis/useGetPost.jsx';
+import { useDoUnLike } from '../apis/useDoUnlike.jsx';
+import { PostLoader } from '../Loaders/PostLoader.jsx';
+import { WithToken } from "../../HOCs/withToken";
+import { useGetUser } from '../apis/useGetUser.jsx';
+import { useDeletePost } from '../apis/useDeletePost.jsx';
+
+
+export const PostsComponent = ({ statePost, setStatePost, user, setUser }) => {
+  const [postRes, getPost] = useGetPost();
+  const [res, doLike] = useDoLike()
+  const [unlikeRes, doUnLike] = useDoUnLike()
+  const [currentPostId, setCurrentPostId] = useState()
+  const [errorMessage, setErrorMessage] = useState('');
+  const [userRes, getUser] = useGetUser()
```

Comment:
what's the use of user context if we are having `getUser` api call in every component?

Reply:
resolved

Reply:
**how is this resolved** ? if you are making getUser call here in `Posts`, in `miniDashboard` etc; the issue is the api call should be invoked once and data should be used in all those components.